client/securedrop_client/config.py

```diff
@@ -2,29 +2,47 @@
 import logging
 import os
 
-logger = logging.getLogger(__name__)
-
-
-class Config:
-    CONFIG_NAME = "config.json"
+try:
+    from qubesdb import QubesDB
+except ImportError:
+    QubesDB = False
 
-    def __init__(self, journalist_key_fingerprint: str) -> None:
-        self.journalist_key_fingerprint = journalist_key_fingerprint
 
-    @classmethod
-    def from_home_dir(cls, sdc_home: str) -> "Config":
-        full_path = os.path.join(sdc_home, Config.CONFIG_NAME)
+logger = logging.getLogger(__name__)
 
-        try:
-            with open(full_path) as f:
-                json_config = json.loads(f.read())
-        except Exception as e:
-            logger.error("Error opening config file at {}: {}".format(full_path, e))
-            json_config = {}
 
-        return Config(
-            journalist_key_fingerprint=json_config.get("journalist_key_fingerprint", None)
-        )
+class Config:
+    """Configuration loaded at runtime from QubesDB (if available) or
+    environment variables."""
+
+    # Mapping of `Config` attributes (keys) to how to look them up (values)
+    # from either QubesDB or the environment.
+    mapping = {
+        "gpg_domain": "QUBES_GPG_DOMAIN",
+        "journalist_key_fingerprint": "SD_SUBMISSION_KEY_FPR",
+    }
+
+    def __init__(self) -> None:
+        """For each attribute, look it up from either QubesDB or the environment
+        and save it to the internal `config` dictionary."""
+        self.config = {}
+
+        for store, lookup in self.mapping.items():
+            if QubesDB:
+                db = QubesDB()
```

Comment:
Is it a performance issue to initialize the QubesDB() object multiple times inside the loop versus outside? (the conditional logic would make it annoying to do outside the loop)

Reply:
I wasn't worried about it for so few items.  But it was easy enough to do with a ternary conditional in 356616a8c1e6814c166a461a4052025427c48031.